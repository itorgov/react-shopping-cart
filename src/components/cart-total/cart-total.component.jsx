import React from 'react';
import {useSelector} from "react-redux";

import {Price} from '../price/price.component';

import styles from './cart-total.module.scss';

export const CartTotal = () => {
    const items = useSelector(state => state.cart.items);
    const total = items.reduce((sum, item) => sum + item.quantity * item.price, 0);

    return (
        <div className={styles.text}>
            Cart total <span className={styles.price}><Price value={total}/></span>
        </div>
    );
};
