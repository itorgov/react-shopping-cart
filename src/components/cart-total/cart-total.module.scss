.text {
  border-top: 0.5rem solid var(--color-text-dark);
  padding-top: 5rem;
  padding-bottom: 0.5rem;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-size: 1.2rem;
  font-weight: 600;
  color: var(--color-text-light);

  @media only screen and (max-width: 43.75em) {
    border: 0;
    padding-top: 1rem;
  }
}

.price {
  letter-spacing: 2px;
  font-weight: 600;
  font-size: 1.6rem;
  color: var(--color-text-dark);
  margin-left: 1rem;
}
