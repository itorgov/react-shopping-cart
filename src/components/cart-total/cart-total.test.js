import React from "react";
import {mount} from "enzyme";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store'

import {CartTotal} from './cart-total.component';

let wrapper;
let store;

beforeEach(() => {
    const mockStore = configureStore();
    const initialState = {
        cart: {
            items: [
                {
                    id: 1,
                    name: "Test name A",
                    price: 4000,
                    quantity: 2,
                },
                {
                    id: 1,
                    name: "Test name B",
                    price: 2000,
                    quantity: 1,
                },
            ],
            note: '',
        }
    };
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><CartTotal /></Provider>);
});

describe('<CartTotal />', () => {
    it('expect to render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('span has to contain correct total value', () => {
        expect(wrapper.find('span').text()).toEqual('$100.00')
    });
});
