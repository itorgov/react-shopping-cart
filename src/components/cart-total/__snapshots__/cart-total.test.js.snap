// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartTotal /> expect to render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CartTotal>
    <div
      className="text"
    >
      Cart total 
      <span
        className="price"
      >
        <Price
          value={10000}
        >
          $100.00
        </Price>
      </span>
    </div>
  </CartTotal>
</Provider>
`;
